tests/api-compatibility/api-compatibility.mk

```diff
@@ -5,9 +5,9 @@ RUNTIME           = mono --debug
 
 MONO_PATH         = $(call GetPath,MonoSource)
 MONO_API_HTML_DIR = $(MONO_PATH)/mcs/tools/mono-api-html
-MONO_API_HTML     = bin/Build$(CONFIGURATION)/mono-api-html.exe
+MONO_API_HTML     = bin/Build$(CONFIGURATION)/lib/xamarin.android/xbuild/Xamarin/Android/mono-api-html.exe
```

Comment:
This *feels* wrong -- we'll need to wait for the PR build to finish to see if it's *actually* wrong -- because the `@(_MonoUtility)` (above) will copy this file into `bin/$(CONFIGURATION)`, *not* `bin/Build$(CONFIGURATION)`.

Furthermore, adding the files to `@(_MonoUtility)` *feels* wrong because it's placing non-installable artifacts into the "installable artifacts" folder.

I think we instead want a `@(_MonoBuildUtility)` item group in `mono-runtimes.targets` which copies the mono archive artifacts into `bin/Builld$(Configuration)`, which would leave *this* line unchanged.

Reply:
[Confirmed (-ish)](https://jenkins.mono-project.com/job/xamarin-android-pr-pipeline-release/762/), it *is* wrong:

```
/Applications/Xcode92.app/Contents/Developer/usr/bin/make run-api-compatibility-tests
csc -out:bin/BuildRelease/lib/xamarin.android/xbuild/Xamarin/Android/mono-api-html.exe ...
error CS2012: Cannot open '/Users/builder/jenkins/workspace/xamarin-android-pr-pipeline-release/xamarin-android/bin/BuildRelease/lib/xamarin.android/xbuild/Xamarin/Android/mono-api-html.exe' for writing -- 'Could not find a part of the path "/Users/builder/jenkins/workspace/xamarin-android-pr-pi
peline-release/xamarin-android/bin/BuildRelease/lib/xamarin.android/xbuild/Xamarin/Android/mono-api-html.exe".'
```

Note that it's wrong because it's still trying to build/create `mono-api-html.exe` -- note the `csc /out:` value! -- but it's trying to create a file into a directory which doesn't exist.  (`csc` can't create all intermediate directories?!)

Reply:
Related: the **make**(1) rules which *create* `mono-api-html` and `mono-api-info` should be *removed*: https://github.com/xamarin/xamarin-android/blob/6797418e3824a1493a05fd5347cc38b645dac194/tests/api-compatibility/api-compatibility.mk#L24-L34

The whole reason we're hitting these files is because `@(_MonoUtility)` copies into `bin/$(Configuration)`, *not* `bin/Build$(Configuration)`, which is where `$(MONO_API_HTML)` is set for and what the `$(MONO_API_HTML)` rule attempts to create:

```
  Copying file from "/Users/builder/jenkins/workspace/xamarin-android-pr-pipeline-release/xamarin-android/external/mono/sdks/out/android-bcl/monodroid_tools/mono-api-html.exe" to "/Users/builder/jenkins/workspace/xamarin-android-pr-pipeline-release/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/mono-api-html.exe".
```

Again, I believe we should introduce a new `@(_MonoBuildUtility)` item group in `mono-runtimes.targets` which copies the specified files into `bin/Build$(Configuration)`, *no subdirectories*.  That would in turn allow *removing* the change on this line (line 8) *and* removing the `$(MONO_API_HTML)` and `$(MONO_API_INFO)` rules.

Reply:
@jonpryor sounds good! I have that patch too (sort of, without the item group). I will update this PR with this approach.